src/shapes/text.class.js

```diff
@@ -879,6 +887,33 @@
       graphemeInfo.angle = info.angle + (this.pathSide ===  'right' ? Math.PI : 0);
     },
 
+    _getGraphemeDirection: function (grapheme, lineIndex, charIndex) {
+      if (this._reExplicitRTL.test(grapheme)) {
+        return 'rtl';
+      }
+      else if (this._reExplicitLTR.test(grapheme)) {
+        return 'ltr';
+      }
+      else {
+        return this._getGraphemeFallbackDirection(lineIndex, charIndex);
+      }
+    },
+
+    _getGraphemeFallbackDirection: function (lineIndex, charIndex) {
+      if (lineIndex === 0 && charIndex === 0) {
+        return this.direction === 'auto' ?
+          fabric.util.getElementStyle(fabric.document.body, 'dir') || 'ltr' :
```

Comment:
i think auto is fine but is not related to bidi.

Reply:
surprising. I thought the 2 are 1.
I shall remove it.

Reply:
https://www.w3.org/International/questions/qa-html-dir#dirauto

Reply:
Once bidi support mostly work, the developer still does have to set rtl and ltr to dictate how the textbox is oriented.
With bidi not working/not done, auto is still usable to detect the user possible preference, right?
I meant those are 2 features that do not intersect. Did i get it right or am i wrong?

Reply:
If i check the links you posted, the direction auto does not seem to infuence the character itself, but the line ( we don't even have different line support ).
So if the textbox is set to auto, the first strong character typed dictates the the line orientation.
Do i read it correctly?

Reply:
Correct